### DISCUSSION TITLE
received AUTHENTICATION_FAILED notify error

### DISCUSSION THREAD

**Discussion by roninmsn** [strongswan/strongswan#1140] [Q&A]
Server: uBuntu 20.04
stongsWan: 5.8.2
Reference: https://www.linode.com/docs/guides/strongswan-vpn-server-install/

![1](https://user-images.githubusercontent.com/35993215/178752575-b849e130-d02b-4367-a13e-2762d74dcc46.jpg)

File: /etc/sysctl.conf
> net.ipv4.ip_forward=1
net.ipv6.conf.all.forwarding=1
net.ipv4.conf.all.accept_redirects = 0
net.ipv6.conf.all.accept_redirects = 0

File: /etc/ipsec.conf
> config setup
        charondebug="ike 1, knl 1, cfg 0, net 1"
        strictcrlpolicy=no
        uniqueids=yes
        cachecrls=no
conn ipsec-ikev2-vpn
      auto=add
      compress=no
      type=tunnel
      keyexchange=ikev2
      fragmentation=yes
      forceencaps=yes
      dpdaction=clear
      dpddelay=300s
      rekey=no
      left=%any
      leftid=@v6.20u.neworld.tk
      leftcert=server.cert.pem
      leftsendcert=always
      leftsubnet=0.0.0.0/0
      right=%any
      rightid=%any
      rightauth=eap-mschapv2
      rightsourceip=10.0.10.0/24
      rightdns= 8.8.8.8,8.8.4.4
      rightsendcert=never
      eap_identity=%identity
      ike=chacha20poly1305-sha512-curve25519-prfsha512,aes256gcm16-sha384-prfsha384-ecp384,aes256-sha1-modp1024,aes128-sha1-modp1024,3des-sha1-modp1024!
      esp=chacha20poly1305-sha512,aes256gcm16-ecp384,aes256-sha256,aes256-sha1,3des-sha1!

File: /etc/ipsec.secrets
> : RSA "/etc/ipsec.d/private/server.key.pem"
test : EAP "123456"

CA:
>ipsec pki --gen --size 4096 --type rsa --outform pem > /etc/ipsec.d/private/ca.key.pem
ipsec pki --self --in /etc/ipsec.d/private/ca.key.pem --type rsa --dn "CN=IKEV2 00 SERVER" --ca --lifetime 3650 --outform pem > /etc/ipsec.d/cacerts/ca.cert.pem

Server:
>ipsec pki --gen --size 4096 --type rsa --outform pem > /etc/ipsec.d/private/server.key.pem
ipsec pki --pub --in /etc/ipsec.d/private/server.key.pem --type rsa | ipsec pki --issue --lifetime 3650 --cacert /etc/ipsec.d/cacerts/ca.cert.pem --cakey /etc/ipsec.d/private/ca.key.pem --dn "CN=v6.20u.neworld.tk" --san="v6.20u.neworld.tk" --flag serverAuth --flag ikeIntermediate --outform pem > /etc/ipsec.d/certs/server.cert.pem

Client:
>ipsec pki --gen --size 4096 --type rsa --outform pem > client.key.pem
ipsec pki --pub --in client.key.pem --type rsa | ipsec pki --issue --lifetime 3650 --cacert /etc/ipsec.d/cacerts/ca.cert.pem --cakey /etc/ipsec.d/private/ca.key.pem --dn "CN=yonghushishi"  --outform pem > client.cert.pem

P12:
>openssl pkcs12 -export -inkey client.key.pem -in client.cert.pem -name "Client"  -certfile /etc/ipsec.d/cacerts/ca.cert.pem -caname "IKEV2 00 SERVER" -out client.cert.p12

**I have tried many times but can't connect. I don't know where the problem is.**

**Comment by strongX509** [comment]
Authentication of the Android client failed on the strongSwan server. Is there a log available from the strongSwan gateway? What did you configure on the Android client as the Client Identity? ````CN=yonghushishi````?

**Comment by roninmsn** [comment]
Windows 11 has the same error.
However, there is no problem putting the same certificates into Windows Server IKEv2.

**Reply by strongX509** [reply]
Without a server log I can't help you debug the authentication problem. And did you set the ````Client identity```` string in the ````Advanced settings```` of the Android strongSwan  configuration?

**Comment by roninmsn** [comment]
I don't know where the server logs are.
Crashes, exactly the same steps as everyone else, but any of my clients just can't connect.

**Reply by Thermi** [reply]
Follow #196 for the logs

**Comment by orkunsenturk** [comment]
bump
